src/components/SingleArticle.jsx

```diff
@@ -3,37 +3,43 @@ import { useEffect, useState } from 'react';
 import { getArticle } from '../utils/api';
 import Comments from './Comments';
 import { getCommentsByArticleId } from '../utils/api';
-import { Spinner } from 'react-bootstrap';
 import { Button } from 'react-bootstrap';
 import { patchVotes } from '../utils/api';
+import CommentAdder from './CommentAdder';
+import { Spinner } from 'react-bootstrap';
 
 const SingleArticle = () => {
     const { article_id } = useParams();
     const [singleArticle, setSingleArticle] = useState({});
     const [comments, setComments] = useState([]);
+    const [isArticleLoading, setIsArticleLoading] = useState(true);
     const [isLoading, setIsLoading] = useState(true);
     const [votes, setVotes] = useState(0);
-    
+    const [isReadOnly, setIsReadOnly] = useState(false);
+
     useEffect(() => {
-        Promise.all([getArticle(article_id), getCommentsByArticleId(article_id)])
-            .then(([articleFromApi, commentsFromApi]) => {
+        getArticle(article_id)
+            .then((articleFromApi) => {
                 setSingleArticle(articleFromApi);
-                setComments(commentsFromApi);
                 setVotes(articleFromApi.votes);
-                setIsLoading(false);
+                setIsArticleLoading(false);
             })
             .catch((err) => {
                 console.log(err);
             })
     }, [article_id]);
 
-    if (isLoading) {
-        return (
-            <Spinner animation="border" role="status">
-              <span className="visually-hidden">Loading...</span>
-            </Spinner>
-          );
-    }
+    useEffect(() => {
+        getCommentsByArticleId(article_id)
+            .then((commentsFromApi) => {
+                setComments(commentsFromApi);
+                setIsLoading(false);
+                setIsReadOnly(false);
+            })
+            .catch((err) => {
```

Comment:
If there is an error we would want the user to have some sort of indication other than a console.log